### ISSUE TITLE
Operation name missing from remote schema queries after upgrading to V2

### ISSUE THREAD

**Issue by jwoo92** [hasura/graphql-engine#8038]
### Version Information

Server Version: v2.0.9
CLI Version (for CLI related issue): v2.0.9

### Environment

OSS

### What is the expected behavior?

Request payload consisting of the following data shape when making a remote schema query:

```json
{
  "operationName": "pingPong",
  "variables": {},
  "query": "query pingPong { ping }"
}
```

### Keywords

Updating to Hasura v2, operationName

### What is the current behavior?

Migrated from v1 to v2 and found queries made to remote schemas are having their operation name stripped from the query. 

Network logs show the client's request including operation name and server logs show the operation name is removed from the query. If I roll back to v1, this issue goes away. It appears to occur only when using v2.

Here is what the payload looks like after upgrading to V2. `operationName` key/value pair and the operation name's value within the `query` string are both stripped away.

```json
{
  "variables": {},
  "query": "query { ping }"
}
```

### How to reproduce the issue?

1. Use Hasura GraphQL Engine v2.0.9
2. Add request body field `operationName` with any string value and/or add an operation name to the query string value for request body field `query`

### Screenshots or Screencast

N/A

### Please provide any traces or logs that could help here.

N/A

### Any possible solutions?

N/A

### Can you identify the location in the source code where the problem exists?

N/A

### If the bug is confirmed, would you be willing to submit a PR?

N/A

**Comment by namoscato** [comment]
We're seeing the same thing on v2.2.0.

@Naveenaidu, @purugupta99 – is there a timeline on a fix here? This is significantly disrupting our observability.

/cc @BenoitRanque

**Comment by BenoitRanque** [comment]
This is currently being worked on and a PR on the private repo exists.
Expect this fix in the next version or the one after that (current is 2.2.0)

**Comment by arcticmatt** [comment]
Fixing this would be great. For now, I'm just passing the operation name as a header 😬 (cc @namoscato)

**Comment by purugupta99** [comment]
@namoscato the PR has been merged, you should be able to use this in coming releases as conveyed by Benoit.

**Comment by ecthiender** [comment]
Fixed via 2e823234f9d25fda013f2f2650194b6c2cffb31a
